numpy/core/function_base.py

```diff
@@ -1,10 +1,24 @@
 from __future__ import division, absolute_import, print_function
 
+import warnings
+import operator
+
 __all__ = ['logspace', 'linspace']
 
 from . import numeric as _nx
 from .numeric import result_type, NaN, shares_memory, MAY_SHARE_BOUNDS, TooHardError
 
+def _index_deprecate(i, stacklevel=2):
```

Comment:
> I'm pretty sure checking for Integral is subtly wrong.

Could you explain? It gets all the numpy and python integers and their subclasses and can't be worse than calling `__index__` to see what happens. It also guarantees int conversions, and ints are what we want, not indexes.